File: ios/Counter.m
//
//  Counter.m
//  NativeModules
//
//  Created by Srikanth Sharma on 09/04/20.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
@interface RCT_EXTERN_MODULE(Counter, NSObject)
  RCT_EXTERN_METHOD(increment)
  RCT_EXTERN_METHOD(getCount: (RCTResponseSenderBlock)callback)
@end

File: ios/Counter.swift
//
//  Counter.swift
//  NativeModules
//
//  Created by Srikanth Sharma on 09/04/20.
//

import Foundation

@objc(Counter)
class Counter: NSObject {
  @objc static var count = 0
  @objc static func requiresMainQueueSetup () -> Bool {
    return true
  }
  @objc func getCount(_ callback: RCTResponseSenderBlock) {
    callback([NSNull(), Counter.count])
  }
  @objc func increment() {
    Counter.count += 1;
    print("Counter is \(Counter.count)")
  }
}
